src/vs/editor/contrib/referenceSearch/browser/referencesWidget.ts

```diff
@@ -212,8 +212,19 @@ class DataSource implements tree.IDataSource {
 	}
 }
 
+/**
+ * RevealedReference wraps an optional OneReference.
+ * This allows RevealedReference to be passed around (i.e. to Controller)
+ * and updates to the reference can be seen.
+ */
+type RevealedReference = { ref: OneReference | undefined };
```

Comment:
Let me try to explain what I am trying to do

1. When progress events are fired, we set a new model and the widget re-renders.
2. Setting a new model implicitly resets state like (a) the nodes that were expanded and (b) which file was selected
3. Currently, the widget owns the model. Open question: who should own the state 2a and 2b. The current implementation has the widget owning this state, but it sounds like you think the Controller should?
4. Regardless of where 2a and 2b is stored, the widget needs to read the state so it can reapply that state after the new model is set. If it is stored in the controller, then the controller would need a getter/setter and the widget would need to save a reference to the controller (it currently doesn't).